astropy_bot/changelog_checker.py

```diff
@@ -18,6 +18,23 @@ def check_changelog_consistency(pr_handler, repo_handler):
 
     statuses = {}
 
+    # Smart skip: Skip when only the changelog is modified; OR
+    # only requires changelog when changes made to non-test files
+    # in directories of interest.
+    modfiles = pr_handler.get_modified_files()
+    if modfiles == [filename]:
+        statuses['changelog'] = {'description': 'Only changelog itself modified',
+                                 'state': 'success'}
+        return statuses
+    has_code_change = any([f.startswith(('astropy/', 'cextern/')) and
+                           'tests/' not in f for f in modfiles])
+    if ((not has_code_change) and (filename not in modfiles) and
+            ('no-changelog-entry-needed' not in pr_handler.labels) and
+            ('Affects-dev' not in pr_handler.labels)):
+        statuses['changelog'] = {'description': 'Changelog not required for these changes',
+                                 'state': 'success'}
+        return statuses
```

Comment:
Some of this logic is a bit too specific to the core package (the bot can be used on any other packages). It's also not 100% clear to me that files outside of astropy and cextern never require a changelog entry (e.g. setup.cfg). I'd suggest therefore just limiting ``has_code_change`` to just checking that the files aren't all tests?

Reply:
Wait, now I am confused. I thought core-specific stuff *goes here* and generic stuff goes to `baldrick`? Where is core-specific stuff supposed to go, then?

Reply:
I will have to think about this... Making this too generic might also render it pretty much useless. One of the original request was to auto add the label if only doc change is detected. To detect doc change in a more generic way is more complicated, so that is why I did it this way for shorted `if` checks. 🤔 🤔 🤔

Reply:
Ping...